src/FakeItEasy/Creation/FakeObjectCreator.cs

```diff
@@ -29,6 +29,30 @@ internal class FakeObjectCreator : IFakeObjectCreator, IMethodInterceptionValida
             IProxyOptions proxyOptions,
             IDummyValueResolver resolver,
             LoopDetectingResolutionContext resolutionContext)
+        {
+            if (!resolutionContext.TryBeginToResolve(typeOfFake))
+            {
+                return CreationResult.FailedToCreateFake(typeOfFake, "Recursive dependency detected. Already resolving " + typeOfFake + '.');
+            }
+
+            var result = this.CreateFakeWithoutCheckingIfWeAlreadyAre(
+                typeOfFake,
+                proxyOptions,
+                resolver,
+                resolutionContext);
+            if (result.WasSuccessful)
+            {
+                resolutionContext.OnSuccessfulResolve(typeOfFake);
+            }
+
+            return result;
+        }
+
+        public CreationResult CreateFakeWithoutCheckingIfWeAlreadyAre(

```

Comment:
The name is a bit of a mouthful... Maybe `CreateFakeWithoutLoopDetection`?

Reply:
I think the doc comments are clear enough. Since it's not public anyway, I think it's OK to rely on them.